server/src/routes/attainable.ts

```diff
@@ -0,0 +1,140 @@
+// SPDX-FileCopyrightText: 2023 The Aalto Grades Developers
+//
+// SPDX-License-Identifier: MIT
+
+import express, { Router } from 'express';
+
+import { addAttainable, updateAttainable } from '../controllers/attainable';
+import { handleInvalidRequestJson } from '../middleware';
+import { controllerDispatcher } from '../middleware/errorHandler';
+
+export const router: Router = Router();
+
+/**
+ * @swagger
+ * definitions:
+ *   AddAttainment:
+ *     type: object
+ *     description: Information for adding a new study attainment.
+ *     properties:
+ *       parentId:
+ *         type: number
+ *         required: false
+ *         description: Optional parent attainment ID to which the study attainment belongs to.
+ *       name:
+ *         type: string
+ *         required: true
+ *         description: Study attainment name.
+ *       executionDate:
+ *         type: string
+ *         required: true
+ *         description: Date when the attainment is completed (e.g. deadline date or exam date).
+ *       expiryDate:
+ *         type: string
+ *         required: true
+ *         description: Date when the attainment expires.
```

Comment:
This should also explain what it means for an attainment to expire. Something like:

> Once an attainment has expired, it is no longer eligible to count as completion for future course instances.